Imported the following commit with some adjustments to match our version:
https://github.com/dojo/dojo/commit/f58c0e19ce509398985908688a768c8fb41191b8
(cf., Bug #38225)
--- a/dojo/sniff.js	2015-04-15 11:55:37.120000000 +0200
+++ b/dojo/sniff.js	2015-04-15 11:46:54.056000000 +0200
@@ -33,7 +33,6 @@
 		}
 		has.add("android", parseFloat(dua.split("Android ")[1]) || undefined);
 		has.add("bb", (dua.indexOf("BlackBerry") >= 0 || dua.indexOf("BB10") >= 0) && parseFloat(dua.split("Version/")[1]) || undefined);
-		has.add("trident", parseFloat(dav.split("Trident/")[1]) || undefined);
 
 		has.add("svg", typeof SVGAngle !== "undefined");
 
@@ -45,19 +44,15 @@
 				has.add("opera", tv >= 9.8 ? parseFloat(dua.split("Version/")[1]) || tv : tv);
 			}
 
-			// Mozilla and firefox
-			if(dua.indexOf("Gecko") >= 0 && !has("khtml") && !has("webkit") && !has("trident")){
-				has.add("mozilla", tv);
-			}
-			if(has("mozilla")){
-				//We really need to get away from this. Consider a sane isGecko approach for the future.
-				has.add("ff", parseFloat(dua.split("Firefox/")[1] || dua.split("Minefield/")[1]) || undefined);
-			}
-
-			// IE
+			var isIE = 0;
 			if(document.all && !has("opera")){
-				var isIE = parseFloat(dav.split("MSIE ")[1]) || undefined;
-
+				// IE < 11
+				isIE = parseFloat(dav.split("MSIE ")[1]) || undefined;
+			}else if(dav.indexOf("Trident")){
+				// IE >= 9
+				isIE = parseFloat(dav.split("rv:")[1]) || undefined;
+			}
+			if(isIE){
 				//In cases where the page has an HTTP header or META tag with
 				//X-UA-Compatible, then it is in emulation mode.
 				//Make sure isIE reflects the desired version.
@@ -71,6 +66,15 @@
 
 				has.add("ie", isIE);
 			}
+			
+			// Mozilla and firefox
+			if(!has("ie") && dua.indexOf("Gecko") >= 0 && !has("khtml") && !has("webkit")){
+				has.add("mozilla", tv);
+			}
+			if(has("mozilla")){
+				//We really need to get away from this. Consider a sane isGecko approach for the future.
+				has.add("ff", parseFloat(dua.split("Firefox/")[1] || dua.split("Minefield/")[1]) || undefined);
+			}
 
 			// Wii
 			has.add("wii", typeof opera != "undefined" && opera.wiiremote);
--- a/dojo/window.js
+++ b/dojo/window.js
@@ -100,7 +100,7 @@ define(["./_base/lang", "./sniff", "./_base/window", "./dom", "./dom-geometry",
 			// reference to the real window object (maybe a copy), so we must fix it as well
 			// We use IE specific execScript to attach the real window reference to
 			// document._parentWindow for later use
-			if(has("ie") && window !== document.parentWindow){
+			if(has("ie") < 9 && window !== document.parentWindow){
 				/*
 				In IE 6, only the variable "window" can be used to connect events (others
 				may be only copies).
